packages/xstate-solid/src/useActor.ts

```diff
@@ -0,0 +1,71 @@
+import type { ActorRef, Event, EventObject, Sender } from 'xstate';
+import type { Accessor } from 'solid-js';
+import { createEffect, createMemo, on, onCleanup } from 'solid-js';
+import { State } from 'xstate';
+import { deriveServiceState } from './deriveServiceState';
+import { createImmutable } from './createImmutable';
+
+type EmittedFromActorRef<
+  TActor extends ActorRef<any>
+> = TActor extends ActorRef<any, infer TEmitted> ? TEmitted : never;
+
+const noop = () => {
+  /* ... */
+};
+
+// Only spread actor snapshot if it is a xstate state class
+const spreadIfStateInstance = <T>(value: T) =>
+  value instanceof State ? { ...value } : value;
+
+export function useActor<TActor extends ActorRef<any>>(
+  actorRef: Accessor<TActor> | TActor
+): [Accessor<EmittedFromActorRef<TActor>>, TActor['send']];
```

Comment:
The `EmittedFromActorRef` isn't true for machine interpreters here - as we use `spreadIfStateInstance`. I think that it would be better to have types reflecting the runtime and that we should introduce something like a `POJOState` here

Reply:
the POJO thing is the last concern I have at the moment

Reply:
Do we just want a type alias for this? The shape of both the State class and what is produced by `speadIfStateInstance` are the same, other than one being class and the other an object. Not sure how we want to go about this

Reply:
I added a `StateObject` alias